app/models/mvi.rb

```diff
@@ -129,9 +129,12 @@ def mvi_add_person
 
   private
 
-  def clear_cache
-    Mvi.delete(user.uuid)
-    @mvi_response = nil
+  def add_ids(response)
+    # set new ids in the profile and recache the response
+    profile.birls_id = response.mvi_codes[:birls_id] if response.mvi_codes[:birls_id].present?
+    profile.participant_id = response.mvi_codes[:participant_id] if response.mvi_codes[:participant_id].present?
```

Comment:
Are `response.mvi_codes[:birls_id]` and `response.mvi_codes[:participant_id]` `nil` when not `present?`
Looking at the parser below, I think they'll always be `nil` or a stingified digit.
So here's a nitpicky syntax suggestion you can feel free to push back on:
```suggestion
    profile.birls_id = response.mvi_codes[:birls_id].presence
    profile.participant_id = response.mvi_codes[:participant_id].presence
```

Reply:
Sure, looks a little cleaner